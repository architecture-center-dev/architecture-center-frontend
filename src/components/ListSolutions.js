import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import Solution from './Solution';

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
}));

export default function ListSolutions({solutions}) {
  const classes = useStyles();
  console.log(solutions);
  return (
    <div className={classes.root}>
      <Grid container spacing={3} justify="center">
        {
          solutions.map(
            ({solution_id,name,description,year_month}) => 
            (<Grid key={solution_id} item ><Solution solution_id={solution_id} name={name} description={description} date={year_month} image="/checkout.jpg" /></Grid>)
          )
        }        
      </Grid>
    </div>
  );
}