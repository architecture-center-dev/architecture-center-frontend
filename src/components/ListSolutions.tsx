import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import Solution from './Solution';

const useStyles = makeStyles(() => ({
  root: {
    flexGrow: 1,
  },
}));

export type Solution = {
  solution_id: string,
  name: string,
  description: string,
  year_month: string,
  big_picture: string
}

export type ListSolutionsProps = {
  solutions: Array<Solution>
}

export default function ListSolutions({ solutions }: ListSolutionsProps) {
  const classes = useStyles();

  return (
    <div className={classes.root} data-testid="list-solutions">
      <Grid container spacing={3} justify="center">
        {
          solutions.map(
            ({ solution_id, name, description, year_month, big_picture }: Solution) =>
              (<Grid key={solution_id} item ><Solution solution_id={solution_id} name={name} description={description} date={year_month} image={big_picture} /></Grid>)
          )
        }
      </Grid>
    </div>
  );
}