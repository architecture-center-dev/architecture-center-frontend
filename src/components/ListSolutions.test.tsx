import { render } from "@testing-library/react";
import * as React from "react";
import ListSolutions, { ListSolutionsProps, Solution } from "./ListSolutions";

describe("<ListSolutions />", () => {
  it("Component should render correctely", async () => {

    const props: ListSolutionsProps = {
      solutions: []
    }

    const { getByTestId } = render(<ListSolutions {...props} />);
    const listSOlutions = getByTestId('list-solutions');

    expect(listSOlutions).toBeTruthy();
    expect(listSOlutions).toMatchSnapshot();
  });

  it("To pass an array of solutions shold be show the same number of <Solutions />", async () => {

    const props: ListSolutionsProps = {
      solutions: [{
        solution_id: "1",
        name: "solution 1",
        description: "description",
        year_month: "ate",
        big_picture: "image"
      },
      {
        solution_id: "2",
        name: "solution 2",
        description: "description",
        year_month: "ate",
        big_picture: "image"
      }]
    }

    const { getByText } = render(<ListSolutions {...props} />);

    props.solutions.forEach((solution: Solution) => {

      expect(getByText(solution.name)).toBeTruthy();
      
    });
  });

});
