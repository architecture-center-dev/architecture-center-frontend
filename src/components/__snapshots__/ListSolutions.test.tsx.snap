// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListSolutions /> Component should render correctely 1`] = `
<div
  class="makeStyles-root-1"
  data-testid="list-solutions"
>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-justify-xs-center"
  />
</div>
`;
